ledger/participant-integration-api/src/main/scala/platform/store/backend/oracle/OracleStorageBackend.scala

```diff
@@ -521,4 +527,72 @@ private[backend] object OracleStorageBackend
 
   private def submittersIsPartyClause(submittersColumnName: String): (String, String) =
     (s"json_equal(json_query($submittersColumnName, '$$'), json_array(", "))")
+
+  override def createDataSource(
+      jdbcUrl: String,
+      dataSourceConfig: DataSourceStorageBackend.DataSourceConfig,
+      connectionInitHook: Option[Connection => Unit],
+  )(implicit loggingContext: LoggingContext): DataSource = {
+    val oracleDataSource = new oracle.jdbc.pool.OracleDataSource
+    oracleDataSource.setURL(jdbcUrl)
+    InitHookDataSourceProxy(oracleDataSource, connectionInitHook.toList)
+  }
+
+  override def aquireImmediately(
+      lockId: DBLockStorageBackend.LockId,
+      lockMode: DBLockStorageBackend.LockMode,
+  )(connection: Connection): Option[DBLockStorageBackend.Lock] = {
+    val oracleLockMode = lockMode match {
+      case DBLockStorageBackend.LockMode.Exclusive => "6" // "DBMS_LOCK.x_mode"
+      case DBLockStorageBackend.LockMode.Shared => "4" // "DBMS_LOCK.s_mode"
+    }
+    SQL"""
+          SELECT DBMS_LOCK.REQUEST(
+            id => ${oracleIntLockId(lockId)},
+            lockmode => #$oracleLockMode,
+            timeout => 0
+          ) FROM DUAL"""
+      .as(get[Int](1).single)(connection) match {
+      case 0 => Some(DBLockStorageBackend.Lock(lockId, lockMode))
+      case 1 => None
+      case 2 => throw new Exception("Aquiring lock caused a deadlock!")
+      case 3 => throw new Exception("Parameter error")
+      case 4 => Some(DBLockStorageBackend.Lock(lockId, lockMode))
+      case 5 => throw new Exception("Illegal lock handle")
```

Comment:
Where are these exceptions logged? And will they carry the context information that this was a Oracle DBMS_LOCK request that failed?

Reply:
The exception carries this information with the stack trace. 
In HaCoordinator it will be logged without stack trace (only on debug level while retying, and on info level if permanently failed). 
This code also runs in InitHookDataProxy: there it is just exception message on info level.
And bubbles up to HikariCP: there
- if hikari initializes it will be an ERROR log with the full stack trace
- if hikari already initialized it will be swallowed whole

On any rate: I was thinking that the exception messages only will point us to the right direction since it is unique to oracle.
what is your suggestion?

Reply:
I would prefix the messages with `Oracle DB Error ${errorNo}: <your explanation>`. This way the message itself says clearly that this is Oracle specific.

I'd also change `Parameter error` to `Parameter error when trying to acquire lock` again with the motivation to make the error message itself more specific wrt the point that raised it.

Reply:
fixed accordingly